salt/modules/yumpkg.py

```diff
@@ -1513,18 +1513,20 @@ def install(name=None,
             else:
                 pkgstr = pkgpath
 
-            # Lambda to trim the epoch from the currently-installed version if
-            # no epoch is specified in the specified version
-            norm_epoch = lambda x, y: x.split(':', 1)[-1] \
-                if ':' not in y \
-                else x
             cver = old_as_list.get(pkgname, [])
             if reinstall and cver:
                 for ver in cver:
-                    ver = norm_epoch(ver, version_num)
-                    if salt.utils.versions.compare(ver1=version_num,
+                    if diff_attr:
+                        # Since diff_attr was provided, the version info
+                        # is a dict.
+                        ver2 = ver['version']
+                    else:
+                        # No diff_attr was provided, so version is directly
+                        # in ver.
+                        ver2 = ver
```

Comment:
@isbm If I remember correctly, memory is not allocated twice, since both vars point to the same place in memory. 😜
Think of it it as an investment in future bug prevention. This code will probably only grow over time. 😄